fastlane/Fastfile

```diff
@@ -42,72 +43,51 @@ end
-  # pod_lib_lint(podspec: "StreamChat.podspec", allow_warnings: true)
-  # pod_lib_lint(podspec: "StreamChatUI.podspec", allow_warnings: true)
-
-  # Bump Podspec Version
-  version_bump_podspec(path: 'StreamChat.podspec', version_number: version_number)
-  version_bump_podspec(path: 'StreamChatUI.podspec', version_number: version_number)
-
-  # Bump XCFramework Version
-  version_bump_podspec(path: 'StreamChat-XCFramework.podspec', version_number: version_number)
-  version_bump_podspec(path: 'StreamChatUI-XCFramework.podspec', version_number: version_number)
-
-  # Stage all changes
-  sh('git add -A')
-
-  UI.user_error!('Not committing changes') unless prompt(text: 'Will commit changes. All looking good?', boolean: true)
-
-  sh("git commit -m 'Bump #{version_number}'")
+  previous_version_number = last_git_tag
+  artifacts_path = File.absolute_path('../StreamChatArtifacts.json')
+  swift_environment_path = File.absolute_path('../Sources/StreamChat/Generated/SystemEnvironment+Version.swift')
+  extra_changes = lambda do |release_version|
+    # Set the framework version on the artifacts
+    artifacts = JSON.parse(File.read(artifacts_path))
+    artifacts[release_version.to_s] = "https://github.com/#{github_repo}/releases/download/#{release_version}/StreamChat-All.zip"
+    File.write(artifacts_path, JSON.dump(artifacts))
+
+    # Set the framework version in SystemEnvironment+Version.swift
+    new_content = File.read(swift_environment_path).gsub!(previous_version_number, release_version)
+    File.open(swift_environment_path, 'w') { |f| f.puts(new_content) }
+  end
 
-  UI.user_error!('Not pushing changes') unless prompt(text: 'Will push changes. All looking good?', boolean: true)
+  release_ios_sdk(
+    version: options[:version],
+    bump_type: options[:type],
+    sdk_names: sdk_names,
+    podspec_names: ['StreamChat', 'StreamChat-XCFramework', 'StreamChatUI', 'StreamChatUI-XCFramework'],
+    github_repo: github_repo,
+    extra_changes: extra_changes,
+    update_version_numbers: true,
+    create_pull_request: true
+  )
+end
 
-  push_to_git_remote(tags: false)
+desc 'Completes an SDK Release'
+lane :publish_release do |options|
+  # Create XCFrameworks
+  xcversion(version: '13.0')
+  clean_products
+  build_xcframeworks
+  compress_frameworks
+  clean_products
 
-  # Open a new PR
-  create_pull_request(
-    api_token: ENV.fetch('GITHUB_TOKEN', nil),
-    repo: github_repo,
-    title: "#{version_number} Release",
-    head: "release/#{version_number}",
-    base: 'main',
-    body: changes.to_s
+  release_ios_sdk(
+    version: options[:version],
+    sdk_names: sdk_names,
+    podspec_names: ['StreamChat', 'StreamChat-XCFramework', 'StreamChatUI', 'StreamChatUI-XCFramework'],
+    github_repo: github_repo,
+    publish_release: true,
+    upload_assets: ['Products/StreamChat.zip', 'Products/StreamChatUI.zip', 'Products/StreamChat-All.zip']
```

Comment:
I wonder if should actually have 2 functions for this. One `release_ios_sdk` and another `publish_ios_sdk` 🤔 My only concern is that this could be error-prone since the way we distinguish release vs publish is a combination of arguments 🤔

Reply:
Let me separate it into two actions, it should not take long. For SwiftUI we will just have the same `release` lane that executes two commands (`release` and `publish`) 🙂

Reply:
@nuno-vieira, done 🎉
Check out the latest commit (and [SwiftUI PR](https://github.com/GetStream/stream-chat-swiftui/pull/316/commits/a1c29f0e9adc8bae2a40c1f47b8a0a9470380421) if you wish).

Reply:
Looks great 👌